Objects/namespaceobject.c

```diff
@@ -43,10 +44,42 @@ namespace_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 static int
 namespace_init(_PyNamespaceObject *ns, PyObject *args, PyObject *kwds)
 {
-    if (PyTuple_GET_SIZE(args) != 0) {
-        PyErr_Format(PyExc_TypeError, "no positional arguments expected");
+    PyObject *arg = NULL;
+    if (!PyArg_UnpackTuple(args, _PyType_Name(Py_TYPE(ns)), 0, 1, &arg)) {
         return -1;
     }
+    if (arg != NULL) {
+        PyObject *dict;
+        int ret = 0;
+        if (PyDict_CheckExact(arg)) {
+            dict = Py_NewRef(arg);
+        }
+        else {
+            dict = PyDict_New();
+            if (dict == NULL) {
+                return -1;
+            }
+            PyObject *func;
+            ret = PyObject_GetOptionalAttr(arg, &_Py_ID(keys), &func);
+            if (ret > 0) {
+                Py_DECREF(func);
+                ret = PyDict_Merge(dict, arg, 1);
+            }
+            else if (ret == 0) {
+                ret = PyDict_MergeFromSeq2(dict, arg, 1);
+            }
```

Comment:
It can be simplified to:
```suggestion
            dict = PyObject_CallOneArg((PyObject *)&PyDict_Type, arg);
            if (dict == NULL) {
                return -1;
            }
```

Reply:
That is very neat.